### PR TITLE
Hook up GST_DEBUG log to visual studio output

### LINKED ISSUE #23700: Hook up GST_DEBUG log to visual studio output
Environment variables aren't a thing that exists in UWP, so we need to add a custom log handler for [GST_DEBUG](https://gstreamer.freedesktop.org/documentation/tutorials/basic/debugging-tools.html?gi-language=c). We should take each logging record received and pass it through to the standard Rust logging infrastructure:
* update to the latest gstreamer-rs version (0.14.4)
* [add a logging callback](https://docs.rs/gstreamer/0.14.4/gstreamer/fn.debug_add_log_function.html) in components/servo/lib.rs after initializing gstreamer
* based on the [level](https://docs.rs/gstreamer/0.14.4/gstreamer/enum.DebugLevel.html) of the logging record received, invoke an [appropriate](https://docs.rs/log/0.4.8/log/#macros) logging macro with a string formatted with the log record metadata

### PR DESCRIPTION
**Pull request by ferjm**
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #23700

<strike>Depends on https://github.com/servo/media/pull/312</strike>

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/24255)

### CODE REVIEW

File: components/servo/lib.rs [hunk 1]
```diff
@@ -130,6 +130,29 @@ mod media_platform {
...
 
+    #[cfg(target_os = "windows")]
+    fn set_gstreamer_log_handler() {
+        use gstreamer::{debug_add_log_function, debug_remove_default_log_function, DebugLevel};
+
+        debug_remove_default_log_function();
+        debug_add_log_function(|_, level, file, function, line, _, message| {
+            let message = format!(
+                "{:?} {:?}:{:?}:{:?} {:?}",
+                level, file, line, function, message
```
Comment (nirbheek) [inline components/servo/lib.rs]: Isn't this missing the debug category? That's quite important for figuring out which part of gstreamer is sending the message. See: https://gitlab.freedesktop.org/nirbheek/gst-uwp-test-app/blob/master/hello-world/MainPage.xaml.cpp#L78

File: components/servo/lib.rs [hunk 2]
```diff
@@ -130,6 +130,34 @@ mod media_platform {
     use super::ServoMedia;
     use servo_media_gstreamer::GStreamerBackend;
 
+    #[cfg(target_os = "windows")]
+    fn set_gstreamer_log_handler() {
+        use gstreamer::{debug_add_log_function, debug_remove_default_log_function, DebugLevel};
+
+        debug_remove_default_log_function();
+        debug_add_log_function(|cat, level, file, function, line, _, message| {
```
Comment (sdroege) [inline components/servo/lib.rs]: You probably also want to include the name of the object in the logs somewhere. The `_` parameter :)

**Pull request merged**